modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/model/SeqexecCircuit.scala

```diff
@@ -259,19 +261,21 @@ class WebSocketHandler[M](modelRW: ModelRW[M, Option[WebSocket]]) extends Action
     case WSConnect(d) =>
       effectOnly(Effect(webSocket(d)).after(d.millis))
 
-    case Connecting(ws) =>
-      updated(Some(ws))
+    case Connecting =>
+      noChange
 
-    case Connected =>
-      effectOnly(Effect.action(AppendToLog("Connected")))
+    case Connected(ws, delay) =>
+      val effect = Effect.action(AppendToLog("Connected"))
+      updated(WebSocketConnection(Ready(ws), delay), effect)
 
     case ConnectionError(e) =>
       effectOnly(Effect.action(AppendToLog(e)))
 
     case ConnectionClosed(d) =>
-      logger.fine("Retry connecting in "+ d)
-      val effect = Effect(Future(WSConnect(d)))
-      updated(None, effect)
+      val next = math.min(60000, math.max(250, value.nextAttempt * 2))
+      logger.fine("Retry connecting in "+ next)
+      val effect = Effect(Future(WSConnect(next)))
+      updated(WebSocketConnection(Pending(), next), effect)
   }
 }
 
```

Comment:
I'm not sure how browsers handle nowadays the sudden lost of a TCP connection. If I unplug the network cable I still need to refresh to get the connection retry (OSX/Firefox). If there were messages unable go through the websocket from the webclient, do you know if the browser closes the connection by itself? Not sure how to test it.